### PR TITLE
Org messages

### PR DESCRIPTION
**Pull request by allaVolkov**
messages moved as constants to <package_name>_msg files
### Checklist
- [x] Code compiles correctly
- [ ] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [x] Formating and linting run locally successfully
- [x] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

**Comment by tal-sapan** [comment]
I have some comments with suggestions for improving the error messages, but I think maybe we should split it to 2 PRs: one that only refactors the error messages to constants (and uses them in tests etc) and one to improve the messages themselves (including change some parameters and maybe removing some unnecessary error wraps). It will be easier to manage, I think. The merge of the first PR will be quicker and the second PR can be more thorough, with less chance of merge conflicts.

File: cmd/module_test.go [hunk 1]
```diff
@@ -48,15 +50,15 @@ var _ = Describe("Commands", func() {
 			ep := dir.Loc{SourcePath: packCmdSrc, TargetPath: packCmdTrg}
 			targetTmpDir := ep.GetTargetTmpDir()
 			err := dir.CreateDirIfNotExist(targetTmpDir)
-			if err != nil {
-				logs.Logger.Error(err)
-			}
-			f, _ := os.Create(filepath.Join(targetTmpDir, "ui5app"))
+			Ω(err).Should(Succeed())
+			f, err := os.Create(filepath.Join(targetTmpDir, "ui5app"))
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply cmd/module_test.go]: done

File: cmd/module_test.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package commands
 import (
 	"bytes"
 	"fmt"
+	"github.com/SAP/cloud-mta-build-tool/internal/artifacts"
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply cmd/module_test.go]: done

File: internal/artifacts/artifacts_msg.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	packMsg                       = `packing the "%s" module...`
+	packFailedOnLocMsg            = `could not pack the "%s" module when initializing the location`
+	packFailedOnCommandsMsg       = `could not pack the "%s" module when getting commands`
+	packFailedOnBuildArtifactMsg  = `could not pack the "%s" module while getting the build artifact`
+	packFailedOnTargetArtifactMsg = `could not pack the "%s" module while getting the build artifact target path`
+	packFailedOnFolderCreationMsg = `could not pack of the "%s" module when creating the "%s" folder`
+	packFailedOnCopyMsg           = `could not pack of the "%s" module when copying the "%s" path to the "%s" path`
+	// PackFailedOnArchMsg - message raised when pack fails during archiving the module
+	PackFailedOnArchMsg = `could not pack of the "%s" module when archiving`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package dir
+
+const (
+	// FolderCreationFailedMsg - message raised when folder creation fails because of the existence file with identical name
+	FolderCreationFailedMsg = `could not create the "%s" folder because a file exists with the same name`
+
+	copyFailedOnGetStatusMsg         = `could not copy the "%s" pattern from the "%s" folder to the "%s" folder when getting the status of the source entry: %s`
+	copyFailedMsg                    = `could not copy the "%s" pattern from the "%s" folder to the "%s" folder when copying the "%s" entry to the "%s" entry`
+	archivingFailedOnCreateFolderMsg = `could not archive when creating the "%s" folder`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package dir
+
+const (
+	// FolderCreationFailedMsg - message raised when folder creation fails because of the existence file with identical name
+	FolderCreationFailedMsg = `could not create the "%s" folder because a file exists with the same name`
+
+	copyFailedOnGetStatusMsg         = `could not copy the "%s" pattern from the "%s" folder to the "%s" folder when getting the status of the source entry: %s`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	copyFailedMsg                    = `could not copy the "%s" pattern from the "%s" folder to the "%s" folder when copying the "%s" entry to the "%s" entry`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	archivingFailedOnCreateFolderMsg = `could not archive when creating the "%s" folder`
+	fileCreationFailedMsg            = `could not create of the "%s"" file failed`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+	copyByPatternFailedOnCreateMsg   = `could not copy the patterns [%s,...] from the "%s" folder to the "%s" folder when creating the target folder`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	copyByPatternFailedOnTargetMsg   = `could not copy the patterns [%s,...] from the "%s" folder to the "%s" folder because the target is not a folder`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+	copyByPatternFailedOnMatchMsg    = `could not copy the "%s" pattern from the "%s" folder to the "%s" folder when getting matching entries`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+	// InitLocFailedOnDescMsg - message raised when location initialization failed on descriptor validation
+	InitLocFailedOnDescMsg = "could not initialize the location when validating descriptor"
+
+	// InitLocFailedOnWorkDirMsg - message raised on getting working directory when initializing location
+	InitLocFailedOnWorkDirMsg = "could not initialize the location when getting working directory"
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+	invalidDescMsg            = `the "%s" descriptor is invalid; expected one of the following values: Dev, Dep`
+
+	copyByPatternMsg    = "copying the patterns [%s,...] from the %s folder to the %s folder"
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/archive_msg.go [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+	skipSymbolicLinkMsg = `copying of the entries from the "%s" folder to the "%s" folder skipped the "%s" entry because its mode is a symbolic link`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/archive_msg.go]: changed

File: internal/archive/fsops.go [hunk 14]
```diff
@@ -46,7 +40,7 @@ func Archive(sourcePath, targetArchivePath string, ignore []string) (e error) {
 	// create folder of archive file if not exists
 	err = CreateDirIfNotExist(filepath.Dir(targetArchivePath))
 	if err != nil {
-		return errors.Wrapf(err, `archiving failed when creating the "%s" folder`, targetArchivePath)
+		return errors.Wrapf(err, archivingFailedOnCreateFolderMsg, targetArchivePath)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/fsops.go]: done

File: internal/archive/fsops.go [hunk 15]
```diff
@@ -212,23 +206,18 @@ func CopyByPatterns(source, target string, patterns []string) error {
...
 	infoTargetDir, err := os.Stat(target)
 	if err != nil {
 		err = CreateDirIfNotExist(target)
 		if err != nil {
-			return errors.Wrapf(err,
-				"copying the patterns [%s,...] from the %s folder to the %s folder failed when creating the target folder",
-				patterns[0], source, target)
+			return errors.Wrapf(err, copyByPatternFailedOnCreateMsg, patterns[0], source, target)
 		}
 		logs.Logger.Infof(`the "%s" folder has been created`, target)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/fsops.go]: added

File: internal/archive/mta_location_test.go [hunk 16]
```diff
@@ -148,11 +151,13 @@ var _ = Describe("Location", func() {
 	It("Fails on descriptor validation", func() {
 		_, err := Location("", "", "xx", os.Getwd)
 		Ω(err).Should(HaveOccurred())
+		Ω(err.Error()).Should(ContainSubstring(fmt.Sprintf(invalidDescMsg, "xx")))
 	})
 	It("Fails on implicit source", func() {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/archive/mta_location_test.go]: changed

File: internal/artifacts/artifacts_msg.go [hunk 17]
```diff
@@ -0,0 +1,77 @@
+package artifacts
+
+const (
+	assemblingMsg = `assembling the MTA project...`
+
+	assemblyFailedOnCopyMsg    = `assembly of the MTA project failed when copying the MTA content`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed to "could not copy MTA artifacts to assemble"

File: internal/artifacts/artifacts_msg.go [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+	assemblyFailedOnMetaMsg    = `assembly of the MTA project failed when generating the meta information`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed to "could not generate the MTA metadata"

File: internal/artifacts/artifacts_msg.go [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+	assemblyFailedOnMtarMsg    = `assembly of the MTA project failed when generating the MTA archive`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed to `could not create the MTA archive`

File: internal/artifacts/artifacts_msg.go [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+	assemblyFailedOnCleanupMsg = `assembly of the MTA project failed when executing cleanup`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed to `could not clean temporary files`

File: internal/artifacts/meta_test.go [hunk 21]
```diff
@@ -172,15 +165,11 @@ cli_version:["x"]
 			ep := dir.Loc{SourcePath: getTestPath("mtahtml5"), TargetPath: getResultPath()}
 			err := generateMeta(&ep, &ep, &ep, &ep, false, "cf")
 			Ω(err).Should(HaveOccurred())
-			Ω(err.Error()).Should(ContainSubstring(`generation of the MTAD file failed when converting types according to the "cf" platform: unmarshalling of the platforms failed`))
+			Ω(err.Error()).Should(ContainSubstring(fmt.Sprintf(genMTADTypeTypeCnvMsg, "cf")))
+			Ω(err.Error()).Should(ContainSubstring(platform.UnmarshalFailedMsg))
 			platform.PlatformConfig = platformConfig
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/meta_test.go]: done

File: internal/artifacts/meta_test.go [hunk 22]
```diff
@@ -172,15 +165,11 @@ cli_version:["x"]
...
 		})
 
-		It("Generate Meta fails on mtad adaptation", func() {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/meta_test.go]: test failed not on adaptation, but on missing path.
the same functionality covered by "Fails on create file for manifest path"

File: internal/artifacts/artifacts_msg.go [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+
+	copyContentFailedOnLocMsg   = `could not copy the MTA content when initializing the deployment descriptor location`
+	copyContentFailedOnParseMsg = `could not copy the MTA content when parsing the %s file`
+	pathNotExistsMsg            = `the "%s" path does not exist in the MTA project location`
+	copyContentFailedMsg        = `could not copy the "%s" MTA content to the "%s" target directory because: %s`
+	copyStartMsg                = `copying the MTA content...`
+	copyDoneMsg                 = `copied "%s"`
+	cleanupFailedMsg            = `; cleanup failed`
+
+	invalidPlatformMsg = `the invalid target platform "%s"; supported platforms are: "cf", "neo", "xsa"`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: done

File: internal/artifacts/artifacts_msg.go [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+	buildFailedOnDepsMsg     = `could not build the "%s" module when processing dependencies`
+	buildFailedOnExecCmdMsg  = `could not build the "%s" module when executing commands`
+	buildResultMsg           = `the build results of the "%s" module will be packed and saved in the "%s" folder`
+
+	packMsg                       = `packing the "%s" module...`
+	packFailedOnLocMsg            = `could not pack the "%s" module when initializing the location`
+	packFailedOnCommandsMsg       = `could not pack the "%s" module when getting commands`
+	packFailedOnBuildArtifactMsg  = `could not pack the "%s" module while getting the build artifact`
+	packFailedOnTargetArtifactMsg = `could not pack the "%s" module while getting the build artifact target path`
+	packFailedOnFolderCreationMsg = `could not pack of the "%s" module when creating the "%s" folder`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed

File: internal/artifacts/module_arch.go [hunk 25]
```diff
@@ -254,9 +248,9 @@ func handleCopyMtaContentFailure(targetLocation string, copiedMtaContents []stri
 	message string, messageArguments []interface{}) error {
 	errCleanup := cleanUpCopiedContent(targetLocation, copiedMtaContents)
 	if errCleanup == nil {
-		return fmt.Errorf(message, messageArguments...)
+		return errors.Errorf(message, messageArguments...)
 	}
-	return fmt.Errorf(message+"; cleanup failed", messageArguments...)
+	return errors.Errorf(message+cleanupFailedMsg, messageArguments...)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/module_arch.go]: done

File: internal/artifacts/project.go [hunk 26]
```diff
@@ -36,19 +35,19 @@ func ExecBuild(makefileTmp, buildProjectCmdSrc, buildProjectCmdTrg, buildProject
 	error := os.Remove(filepath.Join(buildProjectCmdSrc, filepath.FromSlash(makefileTmp)))
 	if err != nil && error != nil {
 		// Remove Makefile_tmp.mta file from directory
-		return errors.Wrapf(err, `execution of the "%s" file failed; removing of the "%s" file failed`, makefileTmp, makefileTmp)
+		return errors.Wrapf(err, execAndRemoveFailedMsg, makefileTmp, makefileTmp)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/project.go]: wrapping added

File: internal/artifacts/artifacts_msg.go [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+	copyContentFailedMsg        = `could not copy the "%s" MTA content to the "%s" target directory because: %s`
+	copyStartMsg                = `copying the MTA content...`
+	copyDoneMsg                 = `copied "%s"`
+	cleanupFailedMsg            = `; cleanup failed`
+
+	invalidPlatformMsg = `the invalid target platform "%s"; supported platforms are: "cf", "neo", "xsa"`
+	adaptationMsg      = `could not adapt the "%s" module path property`
+
+	// UnsupportedPhaseMsg - message raised when phase of mta project build is wrong
+	UnsupportedPhaseMsg     = `the "%s" phase of mta project build is invalid; supported phases: "pre", "post"`
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_msg.go]: changed to capital

File: internal/commands/commands.go [hunk 28]
```diff
@@ -37,7 +37,7 @@ func GetBuilder(module *mta.Module) (string, bool, map[string]string, []string,
 		if builderName == customBuilder {
 			cmdsParam, ok := module.BuildParams[commandsParam]
 			if !ok {
-				logs.Logger.Warn(`the "commands" property is missing in the "custom" builder`)
+				logs.Logger.Warn(missingPropMsg)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/commands/commands.go]: let's wait for some reuse case

File: internal/artifacts/module_arch.go [hunk 29]
```diff
@@ -250,7 +250,7 @@ func handleCopyMtaContentFailure(targetLocation string, copiedMtaContents []stri
 	if errCleanup == nil {
 		return errors.Errorf(message, messageArguments...)
 	}
-	return errors.Errorf(message+cleanupFailedMsg, messageArguments...)
+	return errors.Errorf(message+";"+cleanupFailedMsg, messageArguments...)
```
Comment (tal-sapan) [inline internal/artifacts/module_arch.go]: There should be a space after the `;`, shouldn't there?

File: internal/artifacts/module_arch.go [hunk 30]
```diff
@@ -234,17 +230,15 @@ func copyMtaContent(source, target string, mtaPaths []string, copyInParallel boo
...
-				`"%s" does not exist in the MTA project location`, []interface{}{mtaPath})
+			return handleCopyMtaContentFailure(target, copiedMtaContents, pathNotExistsMsg, []interface{}{mtaPath})
 		}
 		copiedMtaContents = append(copiedMtaContents, mtaPath)
 		targetMtaContent := filepath.Join(target, mtaPath)
 		err := copyMtaContentFromPath(sourceMtaContent, targetMtaContent, mtaPath, target, copyInParallel)
 		if err != nil {
-			return handleCopyMtaContentFailure(target, copiedMtaContents,
-				`error copying the "%s" MTA content to the "%s" target directory because: %s`, []interface{}{mtaPath, source, err.Error()})
+			return handleCopyMtaContentFailure(target, copiedMtaContents, copyContentFailedMsg, []interface{}{mtaPath, source, err.Error()})
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply internal/artifacts/module_arch.go]: This can be done later on in another change

File: internal/artifacts/module_arch.go (hunk 29 above)
Comment (allaVolkov) [reply internal/artifacts/module_arch.go]: done

**Comment by tal-sapan** [comment]
What about the message in project.go line 28 (`generation of the "%s" file failed`)?

**Pull request merged**